modules/adprimeBidAdapter.js

```diff
@@ -0,0 +1,102 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+import * as utils from '../src/utils.js';
+
+const BIDDER_CODE = 'adprime';
+const AD_URL = 'https://delta.adprime.com/?c=o&m=multi';
+
+function isBidResponseValid(bid) {
+  if (!bid.requestId || !bid.cpm || !bid.creativeId ||
+    !bid.ttl || !bid.currency) {
+    return false;
+  }
+  switch (bid.mediaType) {
+    case BANNER:
+      return Boolean(bid.width && bid.height && bid.ad);
+    case VIDEO:
+      return Boolean(bid.vastUrl);
+    case NATIVE:
+      return Boolean(bid.native && bid.native.title && bid.native.image && bid.native.impressionTrackers);
+    default:
+      return false;
+  }
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+
+  isBidRequestValid: (bid) => {
+    return Boolean(bid.bidId && bid.params && !isNaN(parseInt(bid.params.placementId)));
+  },
+
+  buildRequests: (validBidRequests = [], bidderRequest) => {
+    let winTop = window;
+    let location;
+    try {
+      location = new URL(bidderRequest.refererInfo.referer)
+      winTop = window.top;
+    } catch (e) {
+      location = winTop.location;
+      utils.logMessage(e);
+    };
+    let placements = [];
+    let request = {
+      'deviceWidth': winTop.screen.width,
+      'deviceHeight': winTop.screen.height,
+      'language': (navigator && navigator.language) ? navigator.language.split('-')[0] : '',
+      'secure': 1,
+      'host': location.host,
+      'page': location.pathname,
+      'placements': placements
+    };
+    request.language.indexOf('-') != -1 && (request.language = request.language.split('-')[0])
+    if (bidderRequest) {
+      if (bidderRequest.uspConsent) {
+        request.ccpa = bidderRequest.uspConsent;
+      }
+      if (bidderRequest.gdprConsent) {
+        request.gdpr = bidderRequest.gdprConsent
+      }
+    }
+    const len = validBidRequests.length;
+
+    for (let i = 0; i < len; i++) {
+      let bid = validBidRequests[i];
+      let traff = bid.params.traffic || BANNER
+
+      placements.push({
+        placementId: bid.params.placementId,
+        bidId: bid.bidId,
+        sizes: bid.mediaTypes && bid.mediaTypes[traff] && bid.mediaTypes[traff].sizes ? bid.mediaTypes[traff].sizes : [],
```

Comment:
`sizes` will only have the size information for banner media type. The structures for video and native are different.

See the media type [examples](http://prebid.org/dev-docs/publisher-api-reference.html#addAdUnits-MediaTypes)